### PR TITLE
Feature/determine if simulants have SSN

### PR DESCRIPTION
**Pull request by albrja**
### Determines whether simulants have a SSN on initialization.
- *Category*: Feature
- *JIRA issue*: [MIC-3635](https://jira.ihme.washington.edu/browse/MIC-3635)
- *Research reference*: N/A

### Changes and notes
-Changes data scheme of SSN column from a randomly generated number to a bool
-Strips out code to randomly generate SSN and replaces it with a call to a randomness stream to determine which simulants have a SSN
-Assigns a SSN to simulants natively born in the US and most notably newborns in the simulation.

### Verification and Testing
Ran simulation and saw correctly updated SSN column.

### CODE REVIEW

File: src/vivarium_census_prl_synth_pop/components/population.py [hunk 1]
```diff
@@ -140,9 +138,16 @@ def generate_initial_population(self, pop_data: SimulantData) -> None:
             np.round(pop["age"] * 365.25), unit="days"
         )
 
-        # Add Social Security Numbers
-        pop["ssn"] = self.ssn_generator.generate(pop).ssn
-        pop["ssn"] = self.ssn_generator.remove_ssn(pop["ssn"], self.proportion_with_no_ssn)
+        # Deterimne if simulants have SSN
```
Comment (zmbc) [inline src/vivarium_census_prl_synth_pop/components/population.py:141]:
```suggestion
        # Determine if simulants have SSN
```

File: src/vivarium_census_prl_synth_pop/components/population.py [hunk 2]
```diff
@@ -140,9 +138,16 @@ def generate_initial_population(self, pop_data: SimulantData) -> None:
...
+        pop["ssn"] = False
+        # Give native born simulants a SSN  
+        non_natives_idx = pop.loc[~pop["born_in_us"]].index  # Get index for sims born outside of United States
```
Comment (zmbc) [inline src/vivarium_census_prl_synth_pop/components/population.py:144]: This feels like a double negative. Can we not just use `pop.loc[pop["born_in_us"], "ssn"] = True` in the next line, and define this variable just before `ssn_idx`?

**Review by zmbc — COMMENTED** [review]
There are a bunch of formatting changes in the diff -- the CI should be failing but isn't, see https://github.com/ihmeuw/vivarium_census_prl_synth_pop/pull/108 -- maybe you are running a different version of black or using a custom configuration?

File: src/vivarium_census_prl_synth_pop/components/population.py [hunk 3]
```diff
@@ -140,9 +138,18 @@ def generate_initial_population(self, pop_data: SimulantData) -> None:
...
-        pop["ssn"] = self.ssn_generator.generate(pop).ssn
-        pop["ssn"] = self.ssn_generator.remove_ssn(pop["ssn"], self.proportion_with_no_ssn)
+        # Deterimne if simulants have SSN
+        pop["ssn"] = False
+        # Give native born simulants a SSN
+        pop.loc[pop["born_in_us"], "ssn"] = True  # Give simulants born in US a SSN
+        # Choose which non-native simulants get a SSN
+        non_natives_idx = pop.loc[
+            ~pop["born_in_us"]
+        ].index  # Get index for sims born outside of United States
```
Comment (rmudambi) [inline src/vivarium_census_prl_synth_pop/components/population.py:148]: If you have your comments on separate lines they won't screw up the code formatting as much when you run black.

Also, the syntax below is more succinct, does the same thing, and enables you just to get the (non-)native-born index once.
```suggestion
        native_born_idx = pop.index[pop["born_in_us"]]
        # Give simulants born in US a SSN
        pop.loc[native_born_idx, "ssn"] = True
        
        # Choose which non-native simulants get a SSN
```

You'd need to rename `non_natives_idx` below.

**Review by rmudambi — APPROVED** [review]

**Pull request merged**